sdk/js/src/token_bridge/__tests__/integration.ts

```diff
@@ -604,4 +608,93 @@ describe("Integration Tests", () => {
     });
     // TODO: it has increased balance
   });
+  describe("Terra deposit and transfer tokens", () => {
+    test("Tokens transferred can't exceed tokens deposited", (done) => {
+      (async () => {
+        try {
+          const lcd = new LCDClient({
+            URL: TERRA_NODE_URL,
+            chainID: TERRA_CHAIN_ID,
+          });
+          const mk = new MnemonicKey({
+            mnemonic: TERRA_PRIVATE_KEY,
+          });
+          const wallet = lcd.wallet(mk);
+          const gasPrices = await axios
+            .get(TERRA_GAS_PRICES_URL)
+            .then((result) => result.data);
+          // deposit some tokens (separate transactions)
+          let feeEstimate;
+          for (let i = 0; i < 3; i++) {
+            const deposit = new MsgExecuteContract(
+              wallet.key.accAddress,
+              TERRA_TOKEN_BRIDGE_ADDRESS,
+              {
+                deposit_tokens: {},
+              },
+              { ["uluna"]: "1" }
+            );
+            feeEstimate = await lcd.tx.estimateFee(
+              wallet.key.accAddress,
+              [deposit],
+              {
+                memo: "localhost",
+                feeDenoms: ["uluna"],
+                gasPrices,
+              }
+            );
+            const tx = await wallet.createAndSignTx({
+              msgs: [deposit],
+              memo: "localhost",
+              feeDenoms: ["uluna"],
+              gasPrices,
+              fee: feeEstimate,
+            });
+            await lcd.tx.broadcast(tx);
+          }
+          const provider = new ethers.providers.WebSocketProvider(ETH_NODE_URL);
+          const signer = new ethers.Wallet(ETH_PRIVATE_KEY, provider);
+          // attempt to transfer more than we've deposited
+          const transfer = new MsgExecuteContract(
+            wallet.key.accAddress,
+            TERRA_TOKEN_BRIDGE_ADDRESS,
+            {
+              initiate_transfer: {
+                asset: {
+                  amount: "10000",
+                  info: {
+                    native_token: {
+                      denom: "uluna",
+                    },
+                  },
+                },
+                recipient_chain: CHAIN_ID_ETH,
+                recipient: Buffer.from(signer.publicKey).toString("base64"),
+                fee: "0",
+                nonce: Math.round(Math.round(Math.random() * 100000)),
+              },
+            },
+            {}
+          );
+          let error = false;
+          try {
+            await lcd.tx.estimateFee(wallet.key.accAddress, [transfer], {
+              memo: "localhost",
+              feeDenoms: ["uluna"],
+              gasPrices,
+            });
+          } catch (e) {
+            error = e.response.data.error.includes("Overflow: Cannot Sub");
+          }
+          expect(error).toEqual(true);
```

Comment:
can you add a `withdraw_tokens` message at the end here so the tests are repeatable / leave the state the way they found it 🙂 I think it takes the same asset format as `initiate_transfer`

Reply:
For sure, that's a good idea.

Reply:
@evan-gray should be good now